packages/components/src/List/Toolbar/ColumnChooser/hooks/columnChooserManager.hook.js

```diff
@@ -0,0 +1,154 @@
+import { useState } from 'react';
+import cloneDeep from 'lodash/cloneDeep';
+import invariant from 'invariant';
+import flow from 'lodash/flow';
+import { compareOrder } from '../service';
+
+export function isValueCorrect(value, collectionLength) {
+	return Number.isInteger(value) && (value <= collectionLength || value < 0);
+}
+
+export function getOrderItem(order, index, length) {
+	if (index === length - 1) {
+		return order - 1;
+	}
+	return order + 1;
+}
+
+export function matchOrder(value) {
+	return function match({ order }) {
+		return order === value;
+	};
+}
+
+export function incrementColumnOrder(column, index) {
+	return { ...column, order: index + 1 };
+}
+
+export function organiseEditedColumns(collection) {
+	return collection.sort(compareOrder).map(incrementColumnOrder);
```

Comment:
Do the map before sort or you mutate collection.

Reply:
Nop ordering is important here.
But the map is useless. We want to change the reference.